pkg/gencosmosdb/cosmosdb/template.go

```diff
@@ -161,7 +162,7 @@ func (c *templateClient) QueryAll(ctx context.Context, partitionkey string, quer
 }
 
 func (c *templateClient) ChangeFeed(options *Options) TemplateIterator {
-	return &templateChangeFeedIterator{templateClient: c}
+	return &templateChangeFeedIterator{templateClient: c, options: options}
```

Comment:
Interface and the iterator supported options, but we were not passing them to the iterator for some reason.